

The GTY skip makes GGC clean the registered functions wrongly in lto.

Example:
riscv64-unknown-elf-gcc -flto gcc/testsuite/gcc.target/riscv/rvv/base/bug-3.c
-O2 -march=rv64gcv

In file included from bug-3.c:2: internal compiler error: Segmentation fault

gcc/ChangeLog:

        *riscv-vector-builtins.cc (registered_function): Remove skip at
        decl.


0001-RISC-V-Remove-skip-of-decl-in-registered_function.patch

From 62c7fa9d971d321eaf18e0b450e7f058673d0e55 Mon Sep 17 00:00:00 2001
From: Kuan-Lin Chen <rufus@atccpl09.andestech.com>
Date: Mon, 21 Oct 2024 15:21:55 +0800
Subject: [PATCH 1/2] RISC-V: Remove skip of decl in registered_function.

The GTY skip makes GGC clean the registered functions wrongly in lto.

Example:
riscv64-unknown-elf-gcc -flto gcc/testsuite/gcc.target/riscv/rvv/base/bug-3.c
-O2 -march=rv64gcv

In file included from bug-3.c:2: internal compiler error: Segmentation fault

gcc/ChangeLog:

	*riscv-vector-builtins.cc (registered_function): Remove skip at
	decl.
---
 gcc/config/riscv/riscv-vector-builtins.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/riscv/riscv-vector-builtins.cc b/gcc/config/riscv/riscv-vector-builtins.cc
index 41730c483ee..a99437306f8 100644
--- a/gcc/config/riscv/riscv-vector-builtins.cc
+++ b/gcc/config/riscv/riscv-vector-builtins.cc
@@ -79,7 +79,7 @@ public:
   function_instance GTY ((skip)) instance;
 
   /* The decl itself.  */
-  tree GTY ((skip)) decl;
+  tree GTY (()) decl;
 
   /* The overload hash of non-overloaded intrinsic is determined by
      the overload name and argument list. Adding the overload name to
-- 
2.43.5


